Added file with tests

Lizaveta-Kuzniatsova opened the pull request:
Added two tests with [Fact] and one with [Theory].

Lizaveta-Kuzniatsova committed:
Added file with tests

IvanLukyanau commented on LX.PersonalDevelopmentPlan.Tests/DateTimeParsingTests.cs:
```diff
@@ -0,0 +1,53 @@
+        bool parseResult = DateTime.TryParse(dateString, out resultDate);
+
+        //Assert
+        Assert.True(parseResult, "Date string could not be parsed into a DateTime.");
```
`Assert.True(parseResult);`

IvanLukyanau commented on LX.PersonalDevelopmentPlan.Tests/DateTimeParsingTests.cs:
```diff
@@ -0,0 +1,53 @@
+        bool parseResult = DateTime.TryParse(dateString, out resultDate);
+
+        //Assert
+        Assert.True(parseResult, "Date string could not be parsed into a DateTime.");
```
this is the same as prev comment

IvanLukyanau commented on LX.PersonalDevelopmentPlan.Tests/DateTimeParsingTests.cs:
```diff
@@ -0,0 +1,53 @@
+    [Theory]
+    [InlineData("7/10/2024 13:09", 2024, 7, 10, 13, 9, 0)]
+    [InlineData("8/31/2024 23:00", 2024, 8, 31, 23, 0, 0)]
+    public void ParseTwoStrings(string date, int year, int month, int day, int hour, int minute, int second)
```
too much parameter of method: bad

IvanLukyanau commented on LX.PersonalDevelopmentPlan.Tests/DateTimeParsingTests.cs:
```diff
@@ -0,0 +1,53 @@
+        bool parseResult = DateTime.TryParse(dateString, out resultDate);
+
+        //Assert
+        Assert.True(parseResult, $"Date string {dateString} could not be parsed into a DateTime.");
```
already mentioned earlier

Lizaveta-Kuzniatsova committed:
Fix tests naming, remove unnecessary string from Assert.True(), create CreateDateTime() method.

Lizaveta-Kuzniatsova committed:
Create method GetDate to provide sets of data, change [InlineData] attribute to [MemberData].

IvanLukyanau commented on LX.PersonalDevelopmentPlan.Tests/DateTimeParsingTests.cs:
```diff
@@ -0,0 +1,62 @@
+    }
+    [Theory]
+    [MemberData(nameof(GetDate), parameters: 2)]
+    public void Parse_TwoStringsToDateTime_True(string date, DateTime expectedDate)
```
public void Parse_StringToDateTime_ShouldReturnsExpectedDate(string date, DateTime expectedDate)

Lizaveta-Kuzniatsova committed:
Add ITestOutputHelper
Change name of method to Parse_StringToDateTime_ShouldReturnsExpectedDate
Fix DateTime culture formatting

IvanLukyanau commented on LX.PersonalDevelopmentPlan.Tests/TestSettings.cs:
```diff
@@ -0,0 +1,11 @@
+
+public partial class DateTimeParsingTests
+{
+    public static class TestSettings
```
this could be standalone static class and not a part of partial of DateTimeParsingTests

IvanLukyanau commented on LX.PersonalDevelopmentPlan.Tests/DateTimeParsingTests.cs:
```diff
@@ -1,20 +1,32 @@
 namespace LX.PersonalDevelopmentPlan.Tests;
 
-public class DateTimeParsingTests
+public partial class DateTimeParsingTests
```
```suggestion
public class DateTimeParsingTests
```

Lizaveta-Kuzniatsova replied:
fixed

Lizaveta-Kuzniatsova committed:
Remove partial dependency
